keys.hh

```diff
@@ -916,21 +926,7 @@ template <>
 struct fmt::formatter<clustering_key_prefix::with_schema_wrapper> : fmt::formatter<std::string_view> {
     template <typename FormatContext>
     auto format(const clustering_key_prefix::with_schema_wrapper& pk, FormatContext& ctx) const {
-        const auto& [schema, key] = pk;
-        const auto& types = key.get_compound_type(schema)->types();
-        const auto components = key.components(schema);
-        return fmt::format_to(
-            ctx.out(), "{}",
-            fmt::join(boost::make_iterator_range(
-                          boost::make_zip_iterator(boost::make_tuple(types.begin(),
-                                                                     components.begin())),
-                          boost::make_zip_iterator(boost::make_tuple(types.end(),
-                                                                     components.end()))) |
-                      boost::adaptors::transformed([](const auto& type_and_component) {
-                          auto& [type, component] = type_and_component;
-                          return type->to_string(to_bytes(component));
-                      }),
-                      ":"));
+        return ::detail::format_pk(pk, ctx, ::detail::do_validate_utf8::no);
```

Comment:
Not sure why this difference. I think this is just an oversight, we want to validate uft8 for all keys, to not mess up people's terminals with invalid binary output.

Reply:
@bhalevy do you remember why we only validate partition keys?

Reply:
It seems to be just an oversight.
I fixed the problem causing python to reject the logs and there was no apparent problem with clustering keys, just the partition keys (see 389ef9316fde6f97fb2e9b0daf551ff2bc0eb307)

Reply:
@denesb and @bhalevy hi Botond and Benny, how about absorbing the change in 389ef9316fde6f97fb2e9b0daf551ff2bc0eb307 into this PR as another commit to

1. drop `partition_key_to_string()`
2. drop the `::detail::do_validate_utf8`?
3. so we always validate the utf8 in the formatter

?

Reply:
Sounds good to me.

Reply:
oh, i misinterpreted the commit of 389ef9316fde6f97fb2e9b0daf551ff2bc0eb307 and Benny's comment, so the partition keys are already utf8-validated. while clustering keys do not cause any problems at this moment. so i'd prefer keep as it is. as IMHO, the `fmt` related changes are not supposed to change existing behavior without good reasons.

Reply:
I'm fine with not changing this in this PR, although it would make things simpler (would remove the template).
That said, I think we should validate clustering keys in a follow-up. I don't see any reason the clustering keys are guaranteed to not make problems. I think it is more of a case of use not printing clustering keys often combined with accidentally well behaving clustering key values.

Reply:
Agreed :point_up_2:

Reply:
> I'm fine with not changing this in this PR, although it would make things simpler (would remove the template).

i think it'd remove the parameter of `validate_utf8`, but it won't remove the whole template. as `clustering_key_prefix` + `clustering_key_prefix_view` and `partition_key` + `partition_key_view` are different types.

>  That said, I think we should validate clustering keys in a follow-up. I don't see any reason the clustering keys are guaranteed to not make problems. I think it is more of a case of use not printing clustering keys often combined with accidentally well behaving clustering key values.

ack. will create a follow-up change once this one lands.

Reply:
Ok, I'll queue as-is then.

Reply:
> > I'm fine with not changing this in this PR, although it would make things simpler (would remove the template).
> 
> i think it'd remove the parameter of `validate_utf8`, but it won't remove the whole template. as `clustering_key_prefix` + `clustering_key_prefix_view` and `partition_key` + `partition_key_view` are different types.
> 
> > That said, I think we should validate clustering keys in a follow-up. I don't see any reason the clustering keys are guaranteed to not make problems. I think it is more of a case of use not printing clustering keys often combined with accidentally well behaving clustering key values.
> 
> ack. will create a follow-up change once this one lands.

created at #13641